### PR TITLE
Rename the provisioner deployment

### PR DESCRIPTION
**Pull request by anoopjb**
### Multi-Cluster Deployment
In Multi-Cluster Deployment, The Leader Cluster has the provisioner deployment with the name `provisioner`. The helm manages this deployment. This deployment of the provisioner acts as a template and has a replica count set to 0. The Multi-Cluster Deployer then uses this template and creates provisioners in the sister clusters. The provisioner deployment in sister clusters is now called `active-provisioner`.

Deployments in the Leader Cluster,
```
NAME                                                    READY   UP-TO-DATE   AVAILABLE   AGE
interoperator-broker                                    2/2     2            2           174m
interoperator-multiclusterdeployer-controller-manager   0/0     0            0           174m
interoperator-op-apis-app                               2/2     2            2           174m
interoperator-quota-app                                 2/2     2            2           174m
interoperator-scheduler-controller-manager              2/2     2            2           174m
provisioner                                             0/0     0            0           174m
```
Deployments in the Sister Cluster,
```
NAME                 READY   UP-TO-DATE   AVAILABLE   AGE
active-provisioner   1/1     1            1           156m
```
### Single-Cluster Deployment
In Single-Cluster Deployment, the current practice is to edit the provisioner deployment and set the replica to 1. The helm is unaware of this change made by MCD and may cause problems during the upgrade. To mitigate this, now we have a new active-provisioner deployment along with the provisioner deployment.

Deployments in the Leader/Single Cluster,
```
NAME                                                    READY   UP-TO-DATE   AVAILABLE   AGE
active-provisioner                                      1/1     1            1           168m
interoperator-broker                                    2/2     2            2           174m
interoperator-multiclusterdeployer-controller-manager   0/0     0            0           174m
interoperator-op-apis-app                               2/2     2            2           174m
interoperator-quota-app                                 2/2     2            2           174m
interoperator-scheduler-controller-manager              2/2     2            2           174m
provisioner                                             0/0     0            0           174m
```

Feature: HCPCFS-2661

### CODE REVIEW

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-1 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-2 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n is succeeded.

**Comment by anoopjb** [comment]
### Another Approach
In this approach, the provisioner name in the helm chart is changed to `provisioner-template` template.  The Multi Cluster Deployer creates the provisioner deployment as `provisioner`.

The Leader Cluster after upgrade,
```
$ kubectl -n interoperator get deploy

NAME                                                    READY   UP-TO-DATE   AVAILABLE   AGE
interoperator-broker                                    2/2     2            2           64m
interoperator-multiclusterdeployer-controller-manager   2/2     2            2           64m
interoperator-op-apis-app                               2/2     2            2           64m
interoperator-quota-app                                 2/2     2            2           64m
interoperator-scheduler-controller-manager              2/2     2            2           64m
provisioner                                             2/2     2            2           64m
provisioner-template                                    0/0     0            0           13m
```

The Sister Cluster after upgrade,
```
$ kubectl -n interoperator get deploy

NAME          READY   UP-TO-DATE   AVAILABLE   AGE
provisioner   2/2     2            2           65m
```

**Comment by vivekzhere** [comment]
This approach looks better. Some UTs are failing

**Comment by subhankarc** [comment]
@anoopjb Really a much better and cost effective approach! Liked it.

**Review by subhankarc — APPROVED** [review]

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-2 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-1 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n is succeeded.

File: interoperator/pkg/constants/constants.go [hunk 1]
```diff
@@ -13,9 +13,10 @@ const (
 	LastOperationKey                      = "interoperator.servicefabrik.io/lastoperation"
 	ErrorThreshold                        = 10
 
-	ConfigMapName   = "interoperator-config"
-	ConfigMapKey    = "config"
-	ProvisionerName = "provisioner"
+	ConfigMapName         = "interoperator-config"
+	ConfigMapKey          = "config"
+	ProvisionerName       = "provisioner"
+	LeaderProvisionerName = "provisioner-template"
```
Comment (vivekzhere) [inline interoperator/pkg/constants/constants.go:19]: Now a better name for the constant would be `ProvisionerTemplateName`

**Review by vivekzhere — APPROVED** [review]

File: interoperator/pkg/constants/constants.go (hunk 1 above)
Comment (anoopjb) [reply interoperator/pkg/constants/constants.go]: Yes. `ProvisionerTemplateName` sounds better.
Updated :white_check_mark:

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-2 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-1 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n is succeeded.

**Pull request merged**